src/dataEntities/block.ts

```diff
@@ -50,3 +54,87 @@ export interface ResponderBlock extends IBlockStub {
 }
 
 export interface Block extends IBlockStub, Logs, Transactions, TransactionHashes {}
+
+export class BlockItemStore<TBlock extends IBlockStub> extends StartStopService {
+    private readonly subDb: LevelUp<EncodingDown<string, any>>;
+    constructor(db: LevelUp<EncodingDown<string, any>>) {
+        super("block-item-store");
+        this.subDb = sub(db, `block-item-store`, { valueEncoding: "json" });
+    }
+
+    private itemsByHeight: Map<number, Set<string>> = new Map();
+    private items: Map<string, any> = new Map();
+
+    protected async startInternal() {
+        // load all items from the db
+        for await (const record of this.subDb.createReadStream()) {
+            const { key, value } = (record as any) as { key: string; value: any };
+
+            const i = key.indexOf(":");
+            const height = Number.parseInt(key.substring(0, i));
+            const memKey = key.substring(i + 1);
+
+            const itemsAtHeight = this.itemsByHeight.get(height);
+            if (itemsAtHeight) itemsAtHeight.add(memKey);
+            else this.itemsByHeight.set(height, new Set([memKey]));
+            this.items.set(memKey, value);
+        }
+    }
+    protected async stopInternal() {}
+
+    public async putBlockItem(blockHeight: number, blockHash: string, itemKey: string, item: any) {
```

Comment:
I know you have another ticket that talks about type safety. Although this isnt a perfect solution to type-safety, I think a very quick and easy fix would be to just add some utility functions like:
```
    private static readonly KEY_ANCHOR_STATE = "ANCHOR_STATE";
    public async putAnchorState(componentName: string, blockNumber: number, blockHash: string, anchorState: AnchorState) {
        await this.putBlockItem(blockNumber, blockHash, `${componentName}:${BlockItemStore.KEY_ANCHOR_STATE}`, anchorState);
    }
    public getAnchorState<TAnchorState>(blockHash: string, componentName: string): TAnchorState {
        return this.getItem(blockHash, `${componentName}:${BlockItemStore.KEY_ANCHOR_STATE}`)
    }
```
You could even make the `putBlockItem` and `getItem` functions private if you like - unless you want to keep them public for testing. I think this would be worth doing on this branch.

Although maybe a better call signature would be:
```
    const anchorState = store.get.anchorState(hash, component.name)
```
since we'll have a lot of getter and putter functions that you could collect on a get/put object inside the store.

Reply:
Done in 1294449, with syntax "store.{block|attached|anchorState|etc.}.get/set".